meta-dts-distro/recipes-dts/dts/dasharo-deploy/dasharo-deploy

```diff
@@ -617,6 +574,128 @@ install() {
   fi
 }
 
+update_ec() {
+  echo "Downloading EC firmware update..."
+  curl --progress-bar -f "$EC_LINK" -o "$EC_UPDATE_FILE"
+  error_check "Cannot access $FW_STORE_URL. Please check your internet connection"
+  echo "Checking sha256sum..."
+  sha256sum --check <(echo "$EC_HASH")
+  error_check "Failed to download EC firmware update"
+
+  echo "Updating EC..."
+  system76_ectool flash "$EC_UPDATE_FILE" &>> $ERR_LOG_FILE
+  error_check "Failed to update EC firmware"
+}
+
+compare_versions() {
+    # return 1 if ver2 > ver1
+    # return 0 otherwise
+
+    local ver1="$1"
+    local ver2="$2"
+
```

Comment:
@macpijan can we check here if ver1 and ver2 match regex?

`if [[ $ver1 =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]] && [[ $ver2 =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]; then`

that should be ok?

Reply:
added https://github.com/Dasharo/meta-dts/pull/29/files?diff=split&w=0#diff-ac1fdd7af29e82bf8d7ca23462952d638564fce3189e36cbb6eebeb48f480d20R630